commands/auto/autonomousaction.py

```diff
@@ -0,0 +1,117 @@
+from typing import Dict, List
+from commands2 import (
+    Command,
+    CommandBase,
+    FunctionalCommand,
+    ParallelCommandGroup,
+    ParallelDeadlineGroup,
+    SequentialCommandGroup,
+    WaitCommand,
+)
+from pathplannerlib import PathPlannerTrajectory
+from wpilib import DataLogManager
+
+from commands.auto.autohelper import trajectoryFromFile
+from commands.auto.followtrajectory import FollowTrajectory
+from commands.resetdrive import ResetDrive
+from subsystems.drivesubsystem import DriveSubsystem
+
+
+class AutonomousRoutine(SequentialCommandGroup):
+    markerMap: Dict[str, Command]
+
+    def __init__(
+        self,
+        drive: DriveSubsystem,
+        name: str,
+        simultaneousCommands: List[Command],
+    ):
+        self.name = name
+        self.markerMap = {  # later todo: actual implementation
```

Comment:
Smh

Reply:
Fr though shouldn't this be done before we merge it?

Reply:
how? the expected pieces required are in other pull requests
those branches can implement the actual mapping of auto markers to their respective parts